packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts

```diff
@@ -46,33 +47,47 @@ export class QueryDataFileService {
     }
   }
 
-  public save(): string {
+  public async save(): Promise<string> {
+    let queryDataSelectedPath = '';
     const fileContent = this.dataProvider.getFileContent(
       this.queryData.records
     );
-    const savedFileName = this.dataProvider.getFileName();
-    const queryDataFilePath = path.join(
-      this.getResultsDirectoryPath(),
-      savedFileName
+    const defaultFileName = this.dataProvider.getFileName();
+    /* queryDataDefaultFilePath will be used as the default options in the save dialog
+        fileName: The name of the soqlFile viewed in the builder
+        path: the same directory as the .soql file text doc.
+    note: directory must exist to show up in save dialog.
+    */
+    const queryDataDefaultFilePath = path.join(
+      path.parse(this.document.uri.path).dir,
+      defaultFileName
     );
 
-    this.createResultsDirectoryIfDoesNotExist();
-    // Save query results to disk
-    fs.writeFileSync(queryDataFilePath, fileContent);
-    this.showSaveSuccessMessage(savedFileName);
-    this.showFileInExplorer(queryDataFilePath);
+    const saveDialogOptions = {
+      defaultUri: vscode.Uri.file(queryDataDefaultFilePath)
+    };
 
-    return queryDataFilePath;
-  }
+    queryDataSelectedPath = await vscode.window
+      .showSaveDialog({
+        ...saveDialogOptions
+      })
```

Comment:
```suggestion
      .showSaveDialog(saveDialogOptions)
```

Not sure if we need to clone the options right after making them, unless they're to be combined with something else.

Reply:
The method doesn't let me just put the variable in. The signature is looking for the property `defaultUri` so thats why I used the spread operator.

If we thinks its better I could do this
```javascript
.showSaveDialog({
        defaultUri: vscode.Uri.file(queryDataDefaultFilePath)
      })
```

Reply:
Really? That's weird, what's the error? And yeah I think that follow up is better 👍

Reply:
oh ha, ya know what? I left in the extra `{}` from the `...` so it was getting `{saveDialogOptions}`. 
I don't think I need the variable anyway, the property name is semantic already.

Reply:
Fixed in the `fix: clean up code with async/await` commit